lib/ansible/modules/cloud/ovirt/ovirt_nics.py

```diff
@@ -188,18 +207,31 @@ def main():
         # search for the NIC:
         auth = module.params.pop('auth')
         connection = create_connection(auth)
-        vms_service = connection.system_service().vms_service()
-
-        # Locate the VM, where we will manage NICs:
-        vm_name = module.params.get('vm')
-        vm = search_by_name(vms_service, vm_name)
-        if vm is None:
-            raise Exception("VM '%s' was not found." % vm_name)
-
-        # Locate the service that manages the virtual machines NICs:
-        vm_service = vms_service.vm_service(vm.id)
-        nics_service = vm_service.nics_service()
-        vmnics_module = VmNicsModule(
+
+        if module.params.get('vm'):
+            vms_service = connection.system_service().vms_service()
+            # Locate the VM, where we will manage NICs:
+            vm_name = module.params.get('vm')
+            vm = search_by_name(vms_service, vm_name)
+            if vm is None:
+                raise Exception("VM '%s' was not found." % vm_name)
+
+            # Locate the service that manages the virtual machines NICs:
+            service = vms_service.vm_service(vm.id)
+            cluster_name = vm.cluster
+
+        elif module.params.get('template'):
+            templates_service = connection.system_service().templates_service()
+            template_name = module.params.get('template')
+            template = search_by_name(templates_service, template_name)
+            if template is None:
+                raise Exception("Template '%s' was not found." % template_name)
+
+            service = templates_service.template_service(template.id)
+            cluster_name = template.cluster
+
+        nics_service = service.nics_service()
+        entitynics_module = EntityNicsModule(
```

Comment:
Hmm I think this can be simpler, something like this:

```python
if module.params.get('template'):
    entity_name = module.params.get('template')
    collection_service = connection.system_service().templates_service()
elif module.params.get('vm'):
    entity_name = module.params.get('vm')
    collection_service = connection.system_service().vms_service()
            
# TODO: We have to modify the search_by_name function to accept raise_error=True/False,
entity = search_by_name(collection_service, entity_name)
if entity is None:
    raise Exception("Vm/Template '%s' was not found." % entity_name)

service = collection_service.service(entity.id)
cluster_id = entity.cluster
```